src/main/java/htsjdk/samtools/util/EdgeReadIterator.java

```diff
@@ -98,13 +98,14 @@ void accumulateSamRecord(SAMRecord rec) {
             // 0-based offset from the aligned position of the first base in the read to the aligned position
             // of the current base.
             final int refOffset = refPos - rec.getAlignmentStart();
-            final int refOffsetEnd = refPos - rec.getAlignmentStart() + alignmentBlock.getLength();
+            final int refOffsetEnd = refOffset + alignmentBlock.getLength();
 
+            final int accumulatorStartAdjust = accumulator.isEmpty() ? 0 : rec.getAlignmentStart() - accumulator.get(0).getPosition();
 
             // Ensure there are AbstractLocusInfos up to and including this position
-            for (int j = accumulator.size(); j <= refOffsetEnd; ++j) {
+            for (int j = accumulator.size() - accumulatorStartAdjust; j <= refOffsetEnd; ++j) {
```

Comment:
It is a little odd to subtract `rec.getAlignmentStart()` from `j` here and then add it back in below `rec.getAlignmentStart() + 1`. I see that the loop end value, `refOffsetEnd`, is also computed using `rec.getAlignmentStart()`. So it seems like there might be a simpler way to express this loop.